File: product-query-service/src/main/java/com/intexsoft/webshop/productqueryservice/exception/HttpStatusException.java
package com.intexsoft.webshop.productqueryservice.exception;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public abstract class HttpStatusException extends RuntimeException {
    private HttpStatus status;
    private LocalDateTime timeStamp;

    public HttpStatusException(String message, HttpStatus status) {
        super(message);
        this.status = status;
        timeStamp = LocalDateTime.now();
    }
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/exception/HttpStatusException.kt
package com.intexsoft.webshop.productservicekt.exception

import org.springframework.http.HttpStatus
import java.time.LocalDateTime

abstract class HttpStatusException(
    var exceptionMessage: String,
    var status: HttpStatus,
) : RuntimeException(exceptionMessage){
    var timeStamp: LocalDateTime = LocalDateTime.now()
}